### DISCUSSION TITLE
Composer: Segmentation fault

### DISCUSSION THREAD

**Discussion by lixworth** [microsoft/WSL#6970] [Q&A]
### Windows Build Number

Microsoft Windows [版本 10.0.19043.985]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.4.72

### Distro Version

Ubuntu 20.04

### Other Software

PHP 7.4.19 (cli) (built: May 22 2021 03:00:14) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies

Composer version 2.0.11 2021-02-24 14:57:23

### Repro Steps

Run ` composer create-project hyperf/hyperf-skeleton cattery `

### Expected Behavior

Create a composer project

### Actual Behavior

Creating a "hyperf/hyperf-skeleton" project at "./cattery"
Segmentation fault

**Comment by therealkenc** [comment]
No repro offered in the OP, but cannot reproduce locally given the following steps taken from a rando site...

```
$ sudo apt install php-cli unzip php-redis php-xml
$ curl -sS https://getcomposer.org/installer -o composer-setup.php
$ sudo php composer-setup.php --install-dir=/usr/local/bin --filename=composer
```
 
Moving over to discussion. On WSL2 that segfault is going to be a userspace thing. You’ll probably get better eyeballs in a `composer` forum or [SO](https://stackoverflow.com/) (or similar). A websearch on "composer segfault" gets some hits but I don't want to speculate.

![image](https://user-images.githubusercontent.com/7049700/119200343-962b6900-ba41-11eb-9a59-09daa6495388.png)
